test/user/user_db_acl_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package user
+
+import (
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hyperledger-labs/orion-server/pkg/constants"
+	"github.com/hyperledger-labs/orion-server/pkg/server/testutils"
+	"github.com/hyperledger-labs/orion-server/pkg/types"
+	"github.com/hyperledger-labs/orion-server/test/setup"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+var baseNodePort uint32
+var basePeerPort uint32
+var portMutex sync.Mutex
+
+func init() {
+	baseNodePort = 6100
+	basePeerPort = 7100
+}
+
+func getPorts(num uint32) (node uint32, peer uint32) {
+	portMutex.Lock()
+	defer portMutex.Unlock()
+
+	node = baseNodePort
+	peer = basePeerPort
+	baseNodePort += num
+	basePeerPort += num
+
+	return
+}
+
+// Scenario:
+func TestUserACLOnDatabase(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(1)
+	setupConfig := &setup.Config{
+		NumberOfServers:     1,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return errors.Errorf("Redirect blocked in test client: url: '%s', referrer: '%s', #via: %d", req.URL, req.Referer(), len(via))
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+	time.Sleep(10 * time.Second)
+
+	s := c.Servers[0]
+
+	createDatabases(t, s, []string{"db1", "db2", "db3"})
+
+	aliceCert, aliceSigner := testutils.LoadTestClientCrypto(t, c.GetUserCertDir(), "alice")
+	createUsers(t, s, []*types.UserWrite{
+		{
+			User: &types.User{
+				Id:          "alice",
+				Certificate: aliceCert.Raw,
+				Privilege: &types.Privilege{
+					DbPermission: map[string]types.Privilege_Access{
+						"db1": types.Privilege_Read,
+						"db2": types.Privilege_ReadWrite,
+					},
+				},
+			},
+		},
+	})
+
+	client, err := s.NewRESTClient(nil)
+	require.NoError(t, err)
+
+	t.Run("read from db1 would succeed", func(t *testing.T) {
```

Comment:
those are missing:
- write to db3 should fail
- write to db2 should succeed

also, update user:
- db privilege
- certificate

Reply:
Done except the certificate part as it will be tested in the subsequent PR.